SolrPhpClient/Apache/Solr/Service.php

```diff
@@ -328,13 +328,13 @@ protected function _sendRawGet($url, $timeout = FALSE)
 			// use the default timeout pulled from default_socket_timeout otherwise
 			stream_context_set_option($this->_getContext, 'http', 'timeout', $this->_defaultTimeout);
 		}
-		
+
 		// Islandora: dump solr query address in debug mode
 		if (variable_get('islandora_solr_debug_mode', 0) && user_access('view islandora solr debug')) {
                   drupal_set_message(l('solr query',$url."&indent=on&debugQuery=true"));
                 }
 		//$http_response_header is set by file_get_contents
-		$response = new Apache_Solr_Response(@file_get_contents($url, false, $this->_getContext), $http_response_header, $this->_createDocuments, $this->_collapseSingleValueArrays);
+		$response = new Apache_Solr_Response(@file_get_contents($url, false, $this->_getContext), @$http_response_header, $this->_createDocuments, $this->_collapseSingleValueArrays);
```

Comment:
Wuhu! so you are editing the Client =) This opens a lot of possibilities ++

Reply:
I wasn't going to, but then I saw that `islandora_solr_debug_mode` block above. Also I have given up replacing this client, easier to fix it in Islandora 8.